podpac/core/cache/zarr_cache.py

```diff
@@ -0,0 +1,263 @@
+# Create a PODPAC node out of a zarr archive
+import podpac
+from podpac.data import Zarr
+from podpac.core.interpolation.selector import Selector
+from podpac.core.node import Node
+from podpac.core.utils import NodeTrait
+
+import shutil
+import numpy as np
+import traitlets as tl
+import zarr
+import uuid
+from math import ceil
+
+class ZarrCache(Node):
+    """
+    A PODPAC Node which uses Zarr archives to cache data from a source node.
+    
+    Attributes
+    ----------
+    source : Node
+        The data source node.
+    zarr_path_data : str
+        The path to the Zarr archive for storing data.
+    zarr_path_bool : str
+        The path to the Zarr archive for storing boolean indicators of data availability.
+    group_data : zarr.hierarchy.Group
+        The Zarr group for storing data.
+    group_bool : zarr.hierarchy.Group
+        The Zarr group for storing boolean indicators of data availability.
+    selector : Selector
+        Selector for selecting coordinates.
+        
+    _z_node : Zarr
+        Zarr node for data.
+    _z_bool : Zarr
+        Zarr node for boolean indicators.
+    _from_cache : bool
+        Flag indicating whether the data was retrieved from the cache.
+    """
+
+    # Public Traits    
+    source = NodeTrait(allow_none=True).tag(attr=True, required=True)
+    zarr_path_data = tl.Unicode().tag(attr=True, required=True)
+    zarr_path_bool = tl.Unicode().tag(attr=True, required=True)
+    group_data = tl.Instance(zarr.hierarchy.Group).tag(attr=True)
+    group_bool = tl.Instance(zarr.hierarchy.Group).tag(attr=True)
+    selector = tl.Instance(Selector, allow_none=True).tag(attr=True)
+    uid = tl.Instance(uuid.UUID, allow_none=True).tag(attr=True)
+    chunks = tl.List(allow_none=True).tag(attr=True)
+
+    # Private Traits
+    _z_node = tl.Instance(Zarr).tag(attr=True)
+    _z_bool = tl.Instance(Zarr).tag(attr=True)
+    _from_cache = tl.Bool(allow_none=True, default_value=False)
+    
+    
+    @tl.default('selector')
+    def _default_selector(self):
+        return Selector(method='nearest')
+    
+    @tl.default('zarr_path_data')
+    def _default_zarr_path_data(self):
+        return f"{podpac.settings.cache_path}/zarr_cache_data_{self.uid}.zarr"
+    
+    @tl.default('zarr_path_bool')
+    def _default_zarr_path_bool(self):
+        return f"{podpac.settings.cache_path}/zarr_cache_bool_{self.uid}.zarr"
```

Comment:
Because of the way Zarr Nodes work, If I have one zarr node for both groups (making them into arrays for the same group), every time I eval the zarr node, it evals both datasets/arrays. I don't necessarily want this, given that *first* I need to eval the Boolean array without evalling the data array, get the data from the server, *then* only eval the data array.

Reply:
Turning the paths for the zarr_data group and zarr_bool array into private variables, and replacing them with a single variable `zarr_path.`

Reply:
That makes a lot of sense. Please throw a comment into the code to explain the rationale.